### PR TITLE
Add ViewRoom posthog event

### PR DESCRIPTION
**Pull request by yostyle**
#624

### CODE REVIEW

**Review by Johennes — APPROVED** [review]
Looks good as far as I can tell but keep in mind that I'm very much out of my depth on Android. I only added the room creation events and screens to help move things along in our team. 😅

File: appnav/src/main/kotlin/io/element/android/appnav/LoggedInFlowNode.kt [hunk 1]
```diff
@@ -238,6 +239,7 @@ class LoggedInFlowNode @AssistedInject constructor(
                         }
                     }
                 } else {
+                    analyticsService.capture(ViewRoom(isDM = room.isDirect))
```
Comment (julioromano) [inline appnav/src/main/kotlin/io/element/android/appnav/LoggedInFlowNode.kt:242]: Perhaps we should put tracking ode in the node that it directly relates with (i.e. not in the parent node).
What if this was in `RoomFlowNode.init{}` ?

**Comment by julioromano** [comment]
@yostyle can you please put this one on hold a bit until we try to understand if we should send it on the messages view or the whole RoomFlow intead?

**Comment by julioromano** [comment]
@yostyle 
I've checked with the iOS folks, they send this event only when actually seeing the timeline.
So, in order to follow along, the most correct place for this would be `MessagesNode` right now.
Sorry for having you change this so many times :(

**Comment by yostyle** [comment]
> @yostyle I've checked with the iOS folks, they send this event only when actually seeing the timeline. So, in order to follow along, the most correct place for this would be `MessagesNode` right now. Sorry for having you change this so many times :(

The good way could be to capture this event in `RoomFlowNode` and add the trigger argument in node inputs.

**Comment by julioromano** [comment]
> > @yostyle I've checked with the iOS folks, they send this event only when actually seeing the timeline. So, in order to follow along, the most correct place for this would be `MessagesNode` right now. Sorry for having you change this so many times :(
> 
> The good way could be to capture this event in `RoomFlowNode` and add the trigger argument in node inputs.

I'd vouch to do without triggers for the moment as complexity will skyrocket otherwise.

**Pull request merged**